octoprint_slicer/static/js/slicer.js

```diff
@@ -499,6 +506,42 @@ $(function() {
             self.renderer.render( self.scene, self.camera );
         };
 
+	self.slicerProperties = ko.observable();
+	self.updateSlicerProperties = function(newSlicersArray) {
+	    if (_.isArray(newSlicersArray) &&
+		newSlicersArray.length > 0) {
+		OctoPrint
+		    .get(OctoPrint.getBlueprintUrl("slicer") +
+			 "slicer/properties")
+		    .then(function (result) {
+			self.slicerProperties(result);
+		    });
+	    } else {
+		self.slicerProperties({});
+	    }
+	}
+	self.slicingViewModel.slicers.subscribe(self.updateSlicerProperties);
+
+	self.sameDevice = ko.computed(function() {
```

Comment:
Thanks.  I will explain.

The code to allow slicing is here:

https://github.com/foosel/OctoPrint/blob/master/src/octoprint/server/api/files.py#L532

If the slicer is running on the same device as octoprint, slicing is disabled while printing so that the slicing will not use up all the CPU and ruin the print.  But if the slicer is on a different machine (maybe the actual slicing is on a different computer) then slicing is possible even while printing.

In the SlicingViewModel, is_printing and is_paused are available but "same_device" is not.  So there needs to be a way for OctoPrint to ask the server which slicers are same device and which aren't.  That is the Blueprint that I added.

It would be better if the response were available from here:

https://github.com/foosel/OctoPrint/blob/master/src/octoprint/server/api/slicing.py#L72

Then we wouldn't need to add an API and there would be less network traffic.  I will ask @foosel if she can add it to the list.  Let's wait with this PR until we get an answer.